corehq/apps/userreports/sql/adapter.py

```diff
@@ -260,6 +261,96 @@ def doc_exists(self, doc):
             return session.query(query.exists()).scalar()
 
 
+class MultiDBSqlAdapter(object):
+
+    mirror_adapter_cls = IndicatorSqlAdapter
+
+    def __init__(self, config, override_table_name=None):
+        self.config = config
+        self.main_adapter = self.mirror_adapter_cls(config, override_table_name)
+        self.all_adapters = []
+        engine_ids = self.config.get_mirrored_engine_ids(settings.SERVER_ENVIRONMENT) + \
+            [self.main_adapter.engine_id]  # include the main primary adapter
+        # different engine_ids could resolve to same DB, so filter them out
+        engine_ids = connection_manager.filter_out_aliases(engine_ids)
+        for engine_id in engine_ids:
+            self.all_adapters.append(self.mirror_adapter_cls(config, override_table_name, engine_id))
+
+    def __getattr__(self, attr):
+        return getattr(self.main_adapter, attr)
+
+    def handle_exception(self, doc, exception):
```

Comment:
a lot of these functions seem redundant give the `__getattr__` method defined on the class. Basically anything that calls the same function of property on the main table config.

Reply:
Sure